pkg/setup/templates/1.13.go

```diff
@@ -0,0 +1,719 @@
+  level = ""
+
+[metrics]
+  address = "127.0.0.1:1338"
+  grpc_histogram = false
+
+[cgroup]
+  path = ""
+
+[plugins]
+  [plugins.cgroups]
+    no_prometheus = false
+  [plugins.cri]
+    stream_server_address = ""
+    stream_server_port = "10010"
+    enable_selinux = false
+    sandbox_image = "k8s.gcr.io/pause:3.1"
+    stats_collect_period = 10
+    systemd_cgroup = false
+    enable_tls_streaming = false
+    [plugins.cri.containerd]
+      snapshotter = "overlayfs"
+      [plugins.cri.containerd.default_runtime]
+        runtime_type = "io.containerd.runtime.v1.linux"
+        runtime_engine = ""
+        runtime_root = ""
+      [plugins.cri.containerd.untrusted_workload_runtime]
+        runtime_type = ""
+        runtime_engine = ""
+        runtime_root = ""
+    [plugins.cri.cni]
+      bin_dir = "{{.RootABSPath}}/bin"
+      conf_dir = "{{.RootABSPath}}/net.d"
+      conf_template = ""
+    [plugins.cri.registry]
+      [plugins.cri.registry.mirrors]
+        [plugins.cri.registry.mirrors."docker.io"]
+          endpoint = ["https://registry-1.docker.io"]
+  [plugins.diff-service]
+    default = ["walking"]
+  [plugins.linux]
+    shim = "containerd-shim"
+    runtime = "runc"
+    runtime_root = ""
+    no_shim = false
+    shim_debug = false
+  [plugins.scheduler]
+    pause_threshold = 0.02
+    deletion_threshold = 0
+    mutation_threshold = 100
+    schedule_delay = "0s"
+    startup_delay = "100ms"
+`),
+		},
+		{
+			Name:        "kubeconfig-insecure.yaml",
+			Destination: ManifestConfig,
+			Content: []byte(`---
+apiVersion: v1
+kind: Config
+clusters:
+  - cluster:
+      server: http://127.0.0.1:8080
+    name: p8s
+contexts:
+  - context:
+      cluster: p8s
+      user: p8s
+    name: p8s
+current-context: p8s
+users:
+  - name: p8s
+    username: p8s
+`),
+		},
+		{
+			Name:        "audit.yaml",
+			Destination: ManifestConfig,
+			Content: []byte(`---
+apiVersion: audit.k8s.io/v1beta1
+kind: Policy
+rules:
+  - level: Request
+    verbs:
+      - create
+    omitStages:
+      - RequestReceived
+    resources:
+    - group: ""
+      resources:
+        - events
+
+  - level: Metadata
+    omitStages:
+      - RequestReceived
+`),
+		},
+		{
+			Name:        "admission.yaml",
+			Destination: ManifestConfig,
+			Content: []byte(`---
+kind: AdmissionConfiguration
+apiVersion: apiserver.k8s.io/v1alpha1
+plugins:
+- name: EventRateLimit
+  path: eventconfig.yaml
+`),
+		},
+		{
+			Name:        "eventconfig.yaml",
+			Destination: ManifestConfig,
+			Content: []byte(`---
+kind: Configuration
+apiVersion: eventratelimit.admission.k8s.io/v1alpha1
+limits:
+- type: Namespace
+  qps: 50
+  burst: 100
+  cacheSize: 2000
+- type: User
+  qps: 10
+  burst: 50
+`),
+		},
+		{
+			Name:        "kube-controller-manager.yaml",
+			Destination: ManifestAPI,
+			Content: []byte(`---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kube-controller-manager
+  namespace: kube-system
+automountServiceAccountToken: false
```

Comment:
In the logs of the controller manager I see:

```
W0104 14:13:57.716813       1 authentication.go:373] failed to read in-cluster kubeconfig for delegated authentication: failed to read token file "/var/run/secrets/kubernetes.io/serviceaccount/token": open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
W0104 14:13:57.716904       1 authentication.go:235] No authentication-kubeconfig provided in order to lookup client-ca-file in configmap/extension-apiserver-authentication in kube-system, so client certificate authentication won't work.
W0104 14:13:57.716925       1 authentication.go:238] No authentication-kubeconfig provided in order to lookup requestheader-client-ca-file in configmap/extension-apiserver-authentication in kube-system, so request-header client certificate authentication won't work.
W0104 14:13:57.716952       1 authorization.go:177] failed to read in-cluster kubeconfig for delegated authorization: failed to read token file "/var/run/secrets/kubernetes.io/serviceaccount/token": open /var/run/secrets/kubernetes.io/serviceaccount/token: no such file or directory
W0104 14:13:57.716987       1 authorization.go:146] No authorization-kubeconfig provided, so SubjectAccessReview of authorization tokens won't work.
[...]
```

Not sure if this is a result of this option ?

Reply:
Oh this is interesting, I never noticed it before.
I'm going to mount the certificates and a proper kubeconfig to observe to settle this.
The subjectAccessReview is an important feature, but I wonder how the controller manager uses it.